include/nlohmann/json.hpp

```diff
@@ -2212,7 +2218,15 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
             }
         }
 
-        JSON_THROW(type_error::create(306, "cannot use value() with " + std::string(type_name()), *this));
+        JSON_THROW(type_error::create(306, detail::concat("cannot use value() with ", type_name()), this));
+    }
+
+    template<class ValueType, class BasicJsonType, typename std::enable_if<
+                 detail::is_getable<basic_json_t, ValueType>::value, int>::type = 0>
+    JSON_HEDLEY_DEPRECATED(3.10.6)
```

Comment:
Please use `JSON_HEDLEY_DEPRECATED_FOR` and describe what to use instead (see https://nemequ.github.io/hedley/api-reference.html#HEDLEY_DEPRECATED_FOR).

Reply:
The functions should be removed in 4.0.0 without replacement, but I guess this should do:
```
JSON_HEDLEY_DEPRECATED_FOR(3.10.6, basic_json::json_pointer or nlohmann::json_pointer<basic_json::string_t>)
```